managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedCursorTest.java

```diff
@@ -2278,5 +2280,93 @@ public void testReopenMultipleTimes() throws Exception {
         c1 = ledger.openCursor("c1");
     }
 
+    @Test(timeOut = 20000)
```

Comment:
There are 2 tests, one that closes the ledger/factory (simulating graceful shutdown) and  the other that directly opens a new ML factory, simulating a recovery after a crash.